src/circular_buffer.jl

```diff
@@ -18,6 +18,8 @@ end
 
 CircularBuffer(capacity) = CircularBuffer{Any}(capacity)
 
+Base.IndexStyle(::Type{<:CircularBuffer}) = IndexLinear()
+
```

Comment:
`IndexLinear` (as opposed to the default `IndexCartesian` for `AbstractArrays`) allows functions to use an optimized `Base._mapreduce`. This, for example, speeds up `sum`.